roommanager.js

```diff
@@ -0,0 +1,227 @@
+import Player from './player.js'
+import GameRoom from './gameroom.js';
+import { WebSocketServer } from 'ws'
+import { v4 as uuidv4 } from 'uuid';
+
+import { upload_to_blob_storage } from './blobstorage.js';
+
+class RoomManager {
+    constructor(database) {
+        this.database = database
+        this.running_match_id = 1
+        this.game_rooms = {}
+    }
+
+    join_room(ws, join_room_json) {
```

Comment:
I feel like we should try to merge the 2 join paths as much as we can.
Ideally, joining a private match is like joining a special custom queue of which there can be infinite versions, and the queue settings are determined by the person joining it if nobody is in it?
Like custom matches should be a queue named custom_NAMEOFROOM.

When the client joins matchmaking, it just joins the plain "untimed" or whatever.
When the client joins a custom room, it joins "custom_NAMEOFROOM".

Reply:
The the Match List in the client shows a combination of every queue (match and custom) and the active rooms list so you can observe.
Observe should potentially be a totally separate message for simplicity which will go check the room manager to find the right room and join it?

Reply:
this seems natural, will do